Tribler/Core/Modules/cache.py

```diff
@@ -0,0 +1,54 @@
+import codecs
+import os
+
+
+class SimpleCache(object):
+    """
+    This is a cache for recording the keys that we have seen before.
+    """
+    def __init__(self, file_path):
+        self._file_path = file_path
+
+        self._cache_list = list()
+        self._initial_cache_size = 0
+
+    def add(self, key):
+        if not self.has(key):
+            self._cache_list.append(key)
+
+    def has(self, key):
+        return key in self._cache_list
+
+    def load(self):
+        if os.path.exists(self._file_path):
+            try:
+                with codecs.open(self._file_path, 'rb', encoding='utf-8') as f:
+                    lines = f.read()
+
+                self._cache_list = list()
+                lines = lines.split(os.linesep)
+                for line in lines:
+                    if not line:
+                        continue
+                    self._cache_list.append(line)
+                self._initial_cache_size = len(lines)
+            except:
+                return
+        else:
+            f = codecs.open(self._file_path, 'wb', encoding='utf-8')
+            f.close()
+
+    def save(self):
+        if self._initial_cache_size == len(self._cache_list):
+            return
+        try:
+            with codecs.open(self._file_path, 'ab+', encoding='utf-8') as f:
```

Comment:
And also, whats up with the seek stuffs? 
open(self._file_path, "a") should be enough to append to a file right?